chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1745,3 +1748,106 @@ def update_lnet_configuration(self, lnet_configuration_list):
     @property
     def CommandPlan(self):
         return CommandPlan(self._lock_cache, self._job_collection)
+
+    def configure_stratagem(self, stratagem_data):
+        with self._lock:
+            configuration_data = {
+                "state": "unconfigured",
+                "filesystem_id": stratagem_data.get("filesystem"),
+                "interval": stratagem_data.get("interval"),
+                "report_duration": stratagem_data.get("report_duration"),
+                "report_duration_active": stratagem_data.get("report_duration_active"),
+                "purge_duration": stratagem_data.get("purge_duration"),
+                "purge_duration_active": stratagem_data.get("purge_duration_active"),
+            }
+
+            # The filesystem_id may come in as the fs name or the fs id. In terms of storing information in the database, the fs id should always be used.
+            fs_identifier = str(configuration_data.get("filesystem_id"))
+            fs_id = filter(
+                lambda x: str(x.get("id")) == fs_identifier or str(x.get("name")) == fs_identifier,
+                ManagedFilesystem.objects.values("id", "name"),
+            )[0].get("id")
```

Comment:
This is making an assumption that the fs being looked up will always exist.

Reply:
I've abstracted this logic as it's used in a few spots and i've updated it to no longer make this assumption. It will now return back the filesystem id or None.